crates/wast/src/resolve/names.rs

```diff
@@ -1328,32 +1326,35 @@ impl<'a> Module<'a> {
                 };
                 if let FuncKind::Inline { locals, expression } = &mut f.kind {
                     // Resolve (ref T) in locals
-                    for (_, _, valtype) in locals.iter_mut() {
-                        self.resolve_valtype(valtype)?;
+                    for local in locals.iter_mut() {
+                        self.resolve_valtype(&mut local.ty)?;
                     }
 
                     // Build a resolver with local namespace for the function
                     // body
                     let mut resolver = ExprResolver::new(self);
+                    let mut scope = Namespace::default();
 
                     // Parameters come first in the local namespace...
                     if let Some(inline) = &inline {
                         for (id, _, _) in inline.params.iter() {
-                            resolver.locals.register(*id, "local")?;
+                            scope.register(*id, "local")?;
                         }
                     } else if let Some(TypeInfo::Func((params, _))) =
                         self.types.items.get(n as usize).and_then(|i| i.item())
                     {
                         for _ in 0..params.len() {
-                            resolver.locals.register(None, "local")?;
+                            scope.register(None, "local")?;
                         }
                     }
 
                     // .. followed by locals themselves
-                    for (id, _, _) in locals {
-                        resolver.locals.register(*id, "local")?;
+                    for local in locals {
+                        scope.register(local.id, "local")?;
                     }
 
+                    resolver.scopes.push(scope);
```

Comment:
Could you add a debug assert that after `resolve` below the length of `scopes` is 1?